File: Arcade/Intro/3 - Smooth Sailing/11 - isLucky/isLucky.cpp
bool isLucky(int n) {

    std::string str = std::to_string(n);
    int sumLeft = 0;
    int sumRight = 0;
    
    bool lengthIsAnOddNumber = str.length() % 2 != 0;
    if(lengthIsAnOddNumber) return false;
    
    for(int index = 0; index < str.length(); index++)
    {
        bool addingFirstHalh = index < str.length()/2;
        if(addingFirstHalh)
        {
            sumLeft += (int)str.at(index);
        }
        else sumRight += (int)str.at(index);
    }
    
    bool bothValuesAreEqual = sumLeft == sumRight;
    if(bothValuesAreEqual) return true;
    return false;
}

File: Arcade/Intro/3 - Smooth Sailing/11 - isLucky/isLucky.java
boolean isLucky(int n) {
    
    String number = String.valueOf(n);
    int sumLeft = 0;
    int sumRight = 0;
    
    boolean lengthIsAnOddNumber = number.length() % 2 != 0;
    if(lengthIsAnOddNumber) return false;
    
    for(int i = 0; i < number.length(); i++)
    {
        boolean addingTheFirstHalf = i < number.length()/2;
        if(addingTheFirstHalf)
        {
            sumLeft += Character.getNumericValue(number.charAt(i));
        }
        else sumRight += Character.getNumericValue(number.charAt(i));
    }
    
    boolean bothValuesAreEqual = sumLeft == sumRight;
    if(bothValuesAreEqual) return true;
    
    return false;
}
